File: node/multi_model_test.js
// Test multiple models on the same prompt (Node.js)
const OpenAI = require("openai");

const client = new OpenAI({
  baseURL: "https://api.ai24x.com/v1",
  apiKey: "sk-your-key-here",
});

async function main() {
  const prompt = "Explain what an AI gateway is in two sentences.";
  const models = [
    { id: "flash", label: "Fast tier" },
    { id: "pro", label: "Value tier" },
    { id: "vip-gpt56-luna", label: "GPT Luna" },
    { id: "vip-gemini-flash", label: "Gemini Flash" },
  ];

  console.log(Prompt: \n);
  for (const { id, label } of models) {
    const start = Date.now();
    try {
      const response = await client.chat.completions.create({
        model: id,
        messages: [{ role: "user", content: prompt }],
        max_tokens: 150,
      });
      const elapsed = ((Date.now() - start) / 1000).toFixed(2);
      const text = response.choices[0].message.content.slice(0, 60);
      console.log(${id.padEnd(20)}  s  ...);
    } catch (e) {
      console.log(${id.padEnd(20)}  ERROR  );
    }
  }
}

main().catch(console.error);

File: python/multi_model_test.py
#!/usr/bin/env python3
"""Test multiple models on the same prompt for comparison."""

from openai import OpenAI
import time

client = OpenAI(
    base_url="https://api.ai24x.com/v1",
    api_key="sk-your-key-here"
)

prompt = "Explain what an AI gateway is in two sentences."

models = [
    ("flash", "Fast everyday tier"),
    ("pro", "High-value tier"),
    ("vip-gpt5", "GPT-5"),
    ("vip-claude-sonnet", "Claude Sonnet"),
    ("vip-gemini-flash", "Gemini Flash"),
    ("vip-ds-flash", "DeepSeek Flash"),
    ("vip-qwen-max", "Qwen Max"),
]

print(f"Prompt: {prompt}\n")
print(f"{'Model':<25} {'Type':<20} {'Time':<8} {'Response'}")
print("-" * 80)

for model_id, label in models:
    start = time.time()
    try:
        response = client.chat.completions.create(
            model=model_id,
            messages=[{"role": "user", "content": prompt}],
            max_tokens=150
        )
        elapsed = time.time() - start
        text = response.choices[0].message.content[:60].replace("\n", " ")
        print(f"{model_id:<25} {label:<20} {elapsed:<8.2f}s {text}...")
    except Exception as e:
        print(f"{model_id:<25} {label:<20} {'ERROR':<8} {str(e)[:50]}")